File: BE/src/main/java/com/org/workflow/core/exception/WorkFlowException.java
package com.org.workflow.core.exception;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class WorkFlowException extends AbstractException {

  private final ErrorDetail errorDetail;

  public WorkFlowException(ErrorDetail errorDetail) {
    this.errorDetail = errorDetail;
  }

}

File: FastAPI/app/core/exception/work_flow_exception.py
from http import HTTPStatus

from app.core.exception.error_detail import ErrorDetail


class WorkFlowException(Exception):
    error_detail: ErrorDetail
    http_status: HTTPStatus

    def __init__(self, error_detail: ErrorDetail, http_status: HTTPStatus):
        self.http_status = http_status
        self.error_detail = error_detail
